File: _experimental/_elixir/numbers/test/numbers_recursive_test.exs
defmodule NumbersRecursiveTest do
  use ExUnit.Case
  doctest NumbersRecursive

  test "fibonacci of 10" do
    assert NumbersRecursive.fibonacci(10) == 55
  end

  test "factorial of 5" do
    assert NumbersRecursive.factorial(5) == 120
  end

  test "sum of first 5 numbers" do
    assert NumbersRecursive.sum_numbers(5) == 15
  end
end

File: _experimental/_perl/numbers/t/numbers_recursive_test.pl
#!/usr/bin/perl
use strict;
use warnings;

require './lib/numbers_recursive.pl';

use Test::More tests => 2;

is(fibonacci(5), 5, 'fibonacci of 5');

is(factorial(5), 120, 'factorial of 5');

is(sum_numbers(5), 15, 'sum of first 5 numbers');

print "All tests passed.\n";